src/poliastro/core/events.py

```diff
@@ -54,3 +54,33 @@ def eclipse_function(k, u_, r_sec, R_sec, R_primary, umbra=True):
     )
 
     return shadow_function
+
+
+@jit
+def line_of_sight(r1, r2, R):
+    """Calculates the line of sight condition between two position vectors, r1 and r2.
+
+    Parameters
+    ----------
+    r1: ~np.array
+        The position vector of the first object with respect to a central attractor.
+    r2: ~np.array
+        The position vector of the second object with respect to a central attractor.
+    R: float
+        The radius of the central attractor.
+
+    Returns
+    -------
+    delta_theta: float
+        Greater than or equal to zero, if there exists a LOS between two objects
+        located by r1 and r2, else negative.
+
+    """
+    r1_norm = np.linalg.norm(r1)
+    r2_norm = np.linalg.norm(r2)
+
+    theta = np.arccos(np.dot(r1, r2) / r1_norm / r2_norm)
+    theta_1 = np.arccos(R / r1_norm)
```

Comment:
As @Yash-10 pointed out during weekly meeting, if `r1_norm` is lower than `R`, then the argument of the cosine function has a value `>1` and `nan` values are returned.

Reply:
Noted, and handling it in the event function